From 4f9bf0cb0910eea457137102ec460c994068af45 Mon Sep 17 00:00:00 2001
From: Peter Rajnoha <prajnoha@redhat.com>
Date: Fri, 3 Jan 2025 15:26:27 +0100
Subject: [PATCH 18/21] tests: adjust lvresize-xfs tests for recent lvextend
 changes

Because now, we are doing the fsinfo check before extending an LV and if
that check fails, we do not proceed to the LV extension itself and the
lvextend command bails out immediatelly.

(cherry picked from commit e86a75b4fe5873f563c930a575f54143ddca7c7c)
---
 test/shell/lvresize-xfs.sh | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/test/shell/lvresize-xfs.sh b/test/shell/lvresize-xfs.sh
index 87fbf6f9d..17bc6fba3 100644
--- a/test/shell/lvresize-xfs.sh
+++ b/test/shell/lvresize-xfs.sh
@@ -104,13 +104,12 @@ lvremove -f $vg/$lv
 ####################
 
 # lvextend, xfs, active, mounted, --fs resize --fsmode offline
-lvcreate -n $lv -L 300M $vg
+lvcreate -n $lv -L 320M $vg
 mkfs.xfs "$DM_DEV_DIR/$vg/$lv"
 mount "$DM_DEV_DIR/$vg/$lv" "$mount_dir_space"
 df --output=size "$mount_dir_space" |tee df1
 dd if=/dev/zero of="$mount_dir_space/zeros1" bs=1M count=20 oflag=direct
-# xfs_growfs requires the fs to be mounted, so extending the lv is
-# succeeds, then the xfs extend fails because it cannot be done unmounted
+# xfs_growfs requires the fs to be mounted, so the lvextend fails here
 not lvextend --fs resize --fsmode offline -L+20M $vg/$lv
 check lv_field $vg/$lv lv_size "320.00m"
 df | tee dfa
@@ -136,7 +135,7 @@ umount "$mount_dir_space"
 # xfs_growfs requires the fs to be mounted to grow, so --fsmode nochange
 # with an unmounted fs fails
 not lvextend --fs resize --fsmode nochange -L+20M $vg/$lv
-check lv_field $vg/$lv lv_size "380.00m"
+check lv_field $vg/$lv lv_size "360.00m"
 mount "$DM_DEV_DIR/$vg/$lv" "$mount_dir_space"
 df --output=size "$mount_dir_space" |tee df4
 # fs not extended so fs size not changed
@@ -147,7 +146,7 @@ umount "$mount_dir_space"
 # --yes needed because mount changes are required and plain "resize"
 # fsopt did not specify if the user wants to change mount state
 lvextend --yes --fs resize -L+10M $vg/$lv
-check lv_field $vg/$lv lv_size "390.00m"
+check lv_field $vg/$lv lv_size "370.00m"
 mount "$DM_DEV_DIR/$vg/$lv" "$mount_dir_space"
 df --output=size "$mount_dir_space" |tee df5
 not diff df4 df5
@@ -155,7 +154,7 @@ umount "$mount_dir_space"
 
 # lvextend, xfs, active, unmounted, --fs resize_fsadm
 lvextend --fs resize_fsadm -L+10M $vg/$lv
-check lv_field $vg/$lv lv_size "400.00m"
+check lv_field $vg/$lv lv_size "380.00m"
 mount "$DM_DEV_DIR/$vg/$lv" "$mount_dir_space"
 df --output=size "$mount_dir_space" |tee df6
 not diff df5 df6
-- 
2.47.1

